Full screen message shifts up when controls disappear, just before it also disappears

jyavenard opened the pull request:
#### 29adb2b287fb17f11ff8f085c40b543b90166907
<pre>
Full screen message shifts up when controls disappear, just before it also disappears
<a href="https://bugs.webkit.org/show_bug.cgi?id=255777">https://bugs.webkit.org/show_bug.cgi?id=255777</a>
rdar://106225913

Reviewed by Jer Noble.

We had two animations happening in parallel: hiding the banner and hiding
the UI controls.
Both were set to happen for the same duration but in effect this wasn&apos;t
the case.
When the UI controls got hidden, the status bar was set to disappeared
which changed the logical location of the full screen banner and gave the
impression it was shifting up.

Change the anchors of the fullscreen banner to use the one of the parent
view instead.

* Source/WebKit/UIProcess/ios/fullscreen/WKFullScreenViewController.mm:
(-[WKFullScreenViewController loadView]):

Canonical link: <a href="https://commits.webkit.org/263295@main">https://commits.webkit.org/263295@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/af194c36139af0e9d5e6fc61f7b4c4e3f1f3a2cf

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/4180 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/26/builds/4297 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/14/builds/4415 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/5647 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/32/builds/4427 "Built successfully") 
| | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/23/builds/4401 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/16/builds/4262 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/4646 "Passed tests") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/4241 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/15/builds/4414 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/3766 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/5640 "Built successfully") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/39/builds/1910 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/10/builds/3743 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/5924 "Passed tests") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/3737 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/36/builds/3808 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/5328 "Passed tests") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/7/builds/4218 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/17/builds/3418 "Passed tests") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/30/builds/3730 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/3741 "Passed tests") | | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/1022 "Built successfully and passed tests") | [  ~~🛠 watch~~](https://ews-build.webkit.org/#/builders/4/builds/7839 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/31/builds/4026 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

darinadler reviewed (commented):
Seems to me we do not want a two stage animation here, though. We should not let implementation concerns alter the design of the animation. We should first decide what we want it to look like and then implement to make that happen.

But maybe this is a really good design: Can you post a video somewhere of what it looks like?

jyavenard committed:
Full screen message shifts up when controls disappear, just before it also disappears
https://bugs.webkit.org/show_bug.cgi?id=255777
rdar://106225913

Reviewed by Jer Noble.

We had two animations happening in parallel: hiding the banner and hiding
the UI controls.
Both were set to happen for the same duration but in effect this wasn't
the case.
When the UI controls got hidden, the status bar was set to disappeared
which changed the logical location of the full screen banner and gave the
impression it was shifting up.

Change the anchors of the fullscreen banner to use the one of the parent
view instead.

* Source/WebKit/UIProcess/ios/fullscreen/WKFullScreenViewController.mm:
(-[WKFullScreenViewController loadView]):

Canonical link: https://commits.webkit.org/263295@main
